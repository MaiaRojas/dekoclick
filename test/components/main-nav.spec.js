import React from 'react';
import { render, shallow } from 'enzyme';
import sinon from 'sinon';
import MainNav from '../../src/components/main-nav';


describe('<MainNav />', () => {

  it('should print warnings when missing history and firebase props', () => {
    const stub = sinon.stub(console, 'error');
    const component = render(<MainNav />);
		expect(stub.getCalls().length).toBe(2);
    expect(stub.getCall(0).args[0]).toMatchSnapshot();
    expect(stub.getCall(1).args[0]).toMatchSnapshot();
    stub.restore();
	});

  it('should print warnings when missing history.push and firebase.logout props', () => {
    const stub = sinon.stub(console, 'error');
    const component = render(<MainNav history={{}} firebase={{}} />);
		expect(stub.getCalls().length).toBe(2);
    expect(stub.getCall(0).args[0]).toMatchSnapshot();
    expect(stub.getCall(1).args[0]).toMatchSnapshot();
    stub.restore();
  });

  it('should render a <LeftDrawer> with a <List>', () => {
    const component = shallow(
      <MainNav
        history={{ push: () => {} }}
        firebase={{ logout: () => {} }}
      />
    ).dive();

    expect(component.node.type.Naked.name).toBe('LeftDrawer');
    expect(component.node.type.displayName).toBe('withStyles(LeftDrawer)');
    expect(component.node.props.children.type.Naked.name).toBe('List');
    expect(component.node.props.children.type.displayName).toBe('withStyles(List)');
  });

  it('should render a <List> with array of <ListItem>s or <Divider>s', () => {
    const component = shallow(
      <MainNav
        history={{ push: () => {} }}
        firebase={{ logout: () => {} }}
      />
    ).dive();

    expect(component.node.props.children.props.children.length).toBe(6);
    component.node.props.children.props.children.forEach(child => {
      expect(['ListItem', 'Divider'].indexOf(child.type.Naked.name) > -1).toBe(true);
      expect(['withStyles(ListItem)', 'withStyles(Divider)'].indexOf(child.type.displayName) > -1).toBe(true);
    });
  });

});
