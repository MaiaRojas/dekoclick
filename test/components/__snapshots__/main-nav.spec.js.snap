// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainNav /> should print warnings when missing history and firebase props 1`] = `
"Warning: Failed prop type: The prop \`history\` is marked as required in \`MainNav\`, but its value is \`undefined\`.
    in MainNav (created by withStyles(MainNav))
    in withStyles(MainNav)"
`;

exports[`<MainNav /> should print warnings when missing history and firebase props 2`] = `
"Warning: Failed prop type: The prop \`firebase\` is marked as required in \`MainNav\`, but its value is \`undefined\`.
    in MainNav (created by withStyles(MainNav))
    in withStyles(MainNav)"
`;

exports[`<MainNav /> should print warnings when missing history.push and firebase.logout props 1`] = `
"Warning: Failed prop type: The prop \`history.push\` is marked as required in \`MainNav\`, but its value is \`undefined\`.
    in MainNav (created by withStyles(MainNav))
    in withStyles(MainNav)"
`;

exports[`<MainNav /> should print warnings when missing history.push and firebase.logout props 2`] = `
"Warning: Failed prop type: The prop \`firebase.logout\` is marked as required in \`MainNav\`, but its value is \`undefined\`.
    in MainNav (created by withStyles(MainNav))
    in withStyles(MainNav)"
`;
